apps/file/tasks.py

```diff
@@ -0,0 +1,78 @@
+import io
+
+from django.conf import settings
+from PIL import Image
+import requests
+from workers import task
+
+from .libs import upload_file, get_read_url
+from .models import File
+
+
+def _get_size(image, width):
+    """
+    Determines the new image width/height based on given image dimensions and a new width
+
+    """
+    width_percent = width / float(image.size[0])
+    height = float(image.size[1]) * width_percent
+    return (width, height)
+
+
+def _mime_to_format(mime_type):
+    """
+    This is specific to PIL to ensure we use the right format name
+
+    """
+    if mime_type in ('image/jpeg', 'image/jpg'):
+        return 'JPEG'
+
+    if mime_type == 'image/png':
+        return 'PNG'
+
+    raise Exception('unsupported image mime type: {0}'.format(mime_type))
+
+
+def _get_destination(file, size):
+    path = '/'.join(file.link.split('/')[-2:])
+    ext = '.' + path.split('.')[-1]
+    return path.replace(ext, '_{0}{1}'.format(size, ext))
+
+
+@task()
```

Comment:
@geekforbrains should this be scheduled to repeat?

Reply:
@adriaanwm yep will fix, thanks!